src/framework/global/modularity/ioc.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ioc.h"
+
+#include "log.h"
+
+#ifndef NO_QT_SUPPORT
+#include <QtQml>
+
+muse::Injectable::GetContext muse::iocCtxForQmlObject(const QObject* o)
```

Comment:
From the fact that this function returns another function instead of directly a context, I guess that you did this because the context is not yet known at the moment that the "outer" function is called. I guess this is also the reason that you moved something from the DialogView constructor to a method that's called later. 

Which part exactly is not yet known at the point that the outer function is called? I.e. the qmlEngine, the qmlIoc or the cox? And do you know at which point exactly it becomes known? (That's semi-relevant because then we know that before that point we shouldn't use injected things in a class.)

Reply:
The problem there is that at the moment of creating an object, in its constructor, there is still no parent or associated engine. So at this point I can only construct a function to get the context. This does give a limitation that such classes will not be able to use injections in constructor, but this is usually not a problem for such classes.